<launch>

    <arg name="object_labels" value="$(find yolox_models)/config/ycbv_labels.yaml"/>
    <arg name="config" value="$(find clf_object_recognition_yolox)/config/config.yaml"/>
    <arg name="checkpoint" value="$(find yolox_models)/models/ycbv-gdrnpp_bop/model_final.pth" />
    <arg name="exp" value="$(find yolox_models)/config/exp/ycbv-gdrnpp_bop.py"/>

    <rosparam command="load" ns="object_labels" file="$(arg object_labels)"/>
    <arg name="image_topic" default="/xtion/rgb/image_raw"/>

    <node pkg="clf_object_recognition_yolox" type="yolox_recognition_node" name="yolox" output="screen">
        <remap from="~input" to="$(arg image_topic)" />
        <param name="checkpoint" value="$(arg checkpoint)"/>
        <param name="exp" value="$(arg exp)"/>
        <rosparam command="load" file="$(arg config)" />
    </node>

    <!--node name="yolox_detection_3d" pkg="clf_object_recognition_3d" type="new_3d">
		<param name="publish_detections" value="True"/>
        <param name="detect2d_topic" value="/yolox/recognize_from_image"/>
    </node-->
    
    <node name="yolox_detection_3d_pcl" pkg="clf_object_recognition_3d" type="pcl_registration">
		<param name="publish_detections" value="True"/>
		<param name="publish_raw_pcl" value="True"/>
		<param name="publish_clean_pcl" value="True"/>
        <param name="detect2d_topic" value="/yolox/recognize_from_image"/>
    </node>

</launch> 
